librz/core/cmd_debug.c

```diff
@@ -1257,84 +1211,75 @@ static int dump_maps(RzCore *core, int perm, const char *filename) {
 	return ret;
 }
 
-static void cmd_debug_modules(RzCore *core, int mode) { // "dmm"
+static void cmd_debug_current_modules(RzCore *core, RzOutputMode mode) { // "dmm"
 	ut64 addr = core->offset;
 	RzDebugMap *map;
 	RzList *list;
 	RzListIter *iter;
-
-	/* avoid processing the list if the user only wants help */
-	if (mode == '?') {
-	show_help:
-		rz_core_cmd_help(core, help_msg_dmm);
-		return;
+	list = rz_debug_modules_list(core->dbg);
+	rz_list_foreach (list, iter, map) {
+		if (!(addr >= map->addr && addr < map->addr_end)) {
+			continue;
+		}
+		if (mode == RZ_OUTPUT_MODE_STANDARD) {
+			rz_cons_printf("0x%08" PFMT64x " 0x%08" PFMT64x "  %s\n", map->addr, map->addr_end, map->file);
+		} else if (mode == RZ_OUTPUT_MODE_RIZIN) {
+			/* Escape backslashes (e.g. for Windows). */
+			char *escaped_path = rz_str_escape(map->file);
+			char *filtered_name = strdup(map->name);
+			rz_name_filter(filtered_name, 0, true);
+			rz_cons_printf("f mod.%s = 0x%08" PFMT64x "\n",
+				filtered_name, map->addr);
+			rz_cons_printf("oba 0x%08" PFMT64x " %s\n", map->addr, escaped_path);
+			free(escaped_path);
+			free(filtered_name);
+		}
 	}
-	PJ *pj = NULL;
-	if (mode == 'j') {
-		pj = pj_new();
+	rz_list_free(list);
+}
+
+static void cmd_debug_modules(RzCore *core, RzCmdStateOutput *state) { // "dmm"
+	RzDebugMap *map;
+	RzList *list;
+	RzListIter *iter;
+	PJ *pj = state->d.pj;
+	RzOutputMode mode = state->mode;
+	if (mode == RZ_OUTPUT_MODE_JSON) {
 		if (!pj) {
 			return;
 		}
 		pj_a(pj);
 	}
```

Comment:
```suggestion
rz_cmd_state_output_array_start(state);
```

Reply:
If I remove `pj_a(pj)` then JSON objects will not be inside a list, making it an invalid JSON

Reply:
`rz_cmd_state_output_array_start` is doing that.

Reply:
thanks! I missed that